lib/perl/Genome/InstrumentData/AlignmentResult/Speedseq.pm

```diff
@@ -61,13 +63,18 @@ sub _inititalize_revivified_bam {
     return $bam_file;
 }
 
-#Use merged bam for header since we don't have an original per-lane bam
-sub source_bam_path_for_header {
+sub create_bam_header {
     my $self = shift;
-    return $self->get_merged_bam_to_revivify_per_lane_bam;
+
+    return $self->bam_header_path if -s $self->bam_header_path;
+
+    $self->prepare_scratch_sam_file;
```

Comment:
Regard to Adam's comment, an option here is to extract line 591 - 616 of AlignmentResult.pm to a new method prepare_scratch_sam_header_file, and then use $self->prepare_scratch_sam_header_file here.